src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -858,10 +858,10 @@ def endpoints(cls):
                 cls._update_endpoint_params_method_map(
                     func, res_id_params, endpoint_params)
 
-            elif hasattr(func, "_collection_method_"):
+            elif hasattr(func, "collection_method"):
                 cls._update_endpoint_params_collection_map(func, endpoint_params)
 
-            elif hasattr(func, "_resource_method_"):
+            elif hasattr(func, "resource_method"):
```

Comment:
Using underscores for function metadata is a widespread convention, not a PEP8/pylint guideline, but it makes sense, in order to avoid name clashes with properties of real functions...

Reply:
Since `_resource_method_, _collection_method_`  these are protected members of the @RestController class in __init__.py controller, either we need to remove the underscore and make them public methods to access them in the Docs controller Class or we need to disable the pylint error using this `# pylint: disable=protected-access`.

Reply:
Yes, my suggestion is to use double underscore only at the beginning (`__resource_method`), to distinguish them from the Python built-in metadata (`func.__name__` or `func.__doc__`). In any case, it doesn't make sense to have single underscore for a function, as by convention single underscore in Python means 'protected' method, but functions cannot have protected methods as they're not classes.